ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/lexer/LexerDirectoryProcessor.java

```diff
@@ -0,0 +1,36 @@
+package edu.usb.argos.ASTProcessor.lexer;
+
+import org.antlr.v4.runtime.CommonTokenStream;
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public class LexerDirectoryProcessor {
+    private final LexerFileProcessor fileProcessor;
+
+    public LexerDirectoryProcessor() {
+        this.fileProcessor = new LexerFileProcessor();
+    }
+
+    public Map<String, CommonTokenStream> getTokensFromDirectoryByFile(String directoryPath) throws IOException {
+        Map<String, CommonTokenStream> tokensByFile = new HashMap<>();
+
+        try (Stream<Path> paths = Files.walk(Paths.get(directoryPath))) {
+            paths.filter(Files::isRegularFile)
+                    .filter(path -> path.toString().endsWith(".java"))
```

Comment:
This string `.java` would be good if you put it in a constant

Reply:
- In this case, the string ”.java ” is a single-use value in the method, which means that extracting it to a constant may be unnecessary and add complexity without much benefit. The method has a clear purpose: to process .java files in a directory.

- Entering constants for values that are neither reused nor likely to change violates the YAGNI principle. Doing so can lead to an unnecessarily complex design.

Reply:
Done ✅